pkg/proxy/proxy.go

```diff
@@ -107,28 +115,77 @@ func (p *MultiHostReverseProxy) ServeHTTP(w http.ResponseWriter, r *http.Request
 			Msgf("policy %v is not configured", policy)
 	}
 
-	for k := range p.Directors[policy] {
-		if strings.HasPrefix(r.URL.Path, k) && k != "/" {
-			p.Director = p.Directors[policy][k]
-			hit = true
-			p.logger.
-				Debug().
-				Str("policy", policy).
-				Str("prefix", k).
-				Str("path", r.URL.Path).
-				Msg("director found")
+	for i := 0; i < len(config.RouteTypes) && !hit; i++ {
+		routeType := config.RouteTypes[i]
+		var handler func(string, url.URL) bool
+		switch routeType {
+		case config.QueryRoute:
+			handler = p.queryRouteHandler
+		case config.RegexRoute:
+			handler = p.regexRouteHandler
+		case config.PrefixRoute:
+			fallthrough
+		default:
+			handler = p.prefixRouteHandler
+		}
+		for endpoint := range p.Directors[policy][routeType] {
+			if handler(endpoint, *r.URL) {
+				p.Director = p.Directors[policy][routeType][endpoint]
+				hit = true
+				p.logger.
+					Info().
+					Str("policy", policy).
+					Str("prefix", endpoint).
+					Str("path", r.URL.Path).
+					Str("routeType", string(routeType)).
+					Msg("director found")
+				break
```

Comment:
This will only break the inner loop. Could we use a label in here?

Reply:
The outer loop checks if `!hit` but in case of a match we set hit to true so it will exit the loop.

Reply:
aah, too mystical. Still think a label is easier to read

Reply:
Ok :+1: